config/config.go

```diff
@@ -220,6 +222,19 @@ func MergeConfig(to, from *SnapshotterConfig) error {
 		return err
 	}
 
+	// In theory, the value of to.Root is either defaultRootDir or specified by the user.
+	// It is impossible to be empty.
+	if to.Root != "" {
+		logConfig := &to.LoggingConfig
+		if logConfig.LogDir == "" {
+			logConfig.LogDir = filepath.Join(to.Root, logging.DefaultLogDirName)
+		}
+		cacheConfig := &to.CacheManagerConfig
+		if cacheConfig.CacheDir == "" {
+			cacheConfig.CacheDir = filepath.Join(to.Root, "cache")
+		}
+	}
+
```

Comment:
Could we move the above code into `ProcessConfigurations`? I think `MergeConfig` is only a thin layer to merge default configurations and custom configurations. But we can tweak the merged configurations in `ProcessConfigurations`

Reply:
Done. PTAL